proto/encore/parser/schema/v1/schema.proto

```diff
@@ -4,88 +4,141 @@ option go_package = "encr.dev/proto/encore/parser/schema/v1";
+// concrete when there are zero type parameters assigned.
+//
+// For example the Go Code:
+// ```
+// // Set[A] represents our set type
+// type Set[A any] = map[A]struct{}
+// ```
+//
+// Would become:
+// ```
+// _ = &Decl{
+//     id: 1,
+//     name: "Set",
+//     type: &Type{
+//         typ_map: &Map{
+//             key: &Type { typ_type_parameter: ... reference to "A" type parameter below ... },
+//             value: &Type { typ_struct: ... empty struct type ... },
+//         },
+//     },
+//     typeParameters: []*TypeParameter{ { name: "A" } },
+//     doc: "Set[A] represents our set type",
+//     loc: &Loc { ... },
+// }
+// ```
 message Decl {
-  uint32 id = 1;
-  string name = 2;     // type name
-  Type type = 3;
-  string doc = 4;
-  Loc loc = 5;
+  uint32                 id          = 1; // A internal ID which we can refer to this declaration by
+  string                 name        = 2; // The name of the type as assigned in the code
+  Type                   type        = 3; // The underlying type of this declaration
+  repeated TypeParameter type_params = 6; // Any type parameters on this declaration (note; instantiated types used within this declaration would not be captured here)
+  string                 doc         = 4; // The comment block on the type
+  Loc                    loc         = 5; // The location of the declaration within the project
 }
 
+// TypeParameter acts as a place holder for an (as of yet) unknown type in the declaration; the type parameter is
+// replaced with a type argument upon instantiation of the parameterized function or type.
+message TypeParameter {
+  string name = 1; // The identifier given to the type parameter
+}
+
+// Loc is the location of a declaration within the code base
 message Loc {
-  string pkg_path = 1;
-  string pkg_name = 2;
-  string filename = 3;
-  int32 start_pos = 4;
-  int32 end_pos = 5;
-  int32 src_line_start = 6;
-  int32 src_line_end = 7;
-  int32 src_col_start = 8;
-  int32 src_col_end = 9;
+  string pkg_path       = 1; // The package path within the repo (i.e. `/users/signup`)
+  string pkg_name       = 2; // The package name (i.e. `signup`)
+  string filename       = 3; // The file name (i.e. `signup.go`)
+  int32  start_pos      = 4; // The starting index within the file for this node
+  int32  end_pos        = 5; // The ending index within the file for this node
+  int32  src_line_start = 6; // The starting line within the file for this node
+  int32  src_line_end   = 7; // The ending line within the file for this node
+  int32  src_col_start  = 8; // The starting column on the starting line for this node
+  int32  src_col_end    = 9; // The ending column on the ending line for this node
 }
 
+// Named references declaration block by name
 message Named {
-  uint32 id = 1;
+  uint32   id                  = 1; // The `Decl.id` this name refers to
+  repeated Type type_arguments = 2; // The type arguments used to instantiate this parameterised declaration
```

Comment:
As of Go 1.18 Named types also represented instantiated parameterized types - i.e. `MyType[int]` is a named type which has the type arguments of `[int]`